app/classifier.py

```diff
@@ -21,43 +21,64 @@
 
         Parameters:
-            path_to_bert (str): the URL path to a BERT model e.g.
-                    http://tf.google.com/bert_uncased.
-            path_to_classifier (str): the local path to the dir
-                    containing a saved classifier model and its variables.
-            path_to_vocab (str): the local path to a vocab label file to
-                    be used in constructing human-readable topic classification
-                    output.
+            base_classifier_dir (str): the local path to the dir
+                    containing all saved classifier models and their instances.
     """
-
-    def __init__(self, path_to_bert, path_to_classifier, path_to_vocab: str):
-        self.logger = logging.getLogger('app.logger')
-        self.vocab = loadVocab(path_to_vocab)
-        self.classifier = path_to_classifier
-        self.embedder = embedder.Embedder(path_to_bert)
-        self.predictor = tf.contrib.predictor.from_saved_model(self.classifier)
-
-    def classify(self, seq: str) -> [(str, float)]:
+    def __init__(self,
+                 base_classifier_dir: str):
+
+        self.logger = logging.getLogger()
+        self.base_classifier_dir = base_classifier_dir
+
+    def _load_instance(
+            self, relative_path_to_model: str):
+        instances = os.listdir(relative_path_to_model)
+        # pick the latest released instance
+        for i in sorted(instances, reverse=True):
+            with open(os.path.join(
+                    relative_path_to_model, i, "config.json")) as f:
+                d = json.loads(f.read())
+                if d["is_released"]:
+                    self.instance = i
+                    self.instance_config = mc.InstanceConfig(d)
+                    return
+        raise Exception(
+            "No valid instance of model found in %s, instances were %s" % (
+                relative_path_to_model, instances))
+
+    def _load_vocab(self, relative_path_to_vocab: str):
+        with open(relative_path_to_vocab, 'r') as f:
+            self.vocab = f.readlines()
+
+    def classify(self, seq: str, model_name: str) -> [(str, float)]:
         """ classify calculates and returns a particular sequence's topic probability vector.
 
         Parameters:
             seq (str): The text sequence to be classified with this model.
+            model_name (str): the name of the training model (e.g.
+                    UPR_2percent_ps0)
 
         Returns:
             [(str, float)]: The topic probabilty vector in descending order,
                     with topics below the minimum threshold (default=0.4)
                     discarded.
         """
-        embedding = self.embedder.GetEmbedding(seq)
+
+        model_config_path = os.path.join(self.base_classifier_dir, model_name)
+        self._load_instance(model_config_path)
+
+        fq_instance_dir = os.path.join(model_config_path, self.instance)
```

Comment:
fq? I'd prefer longer names, generally.

Reply:
it didn't actually add any value. changed to instance_dir